server/devpi_server/extpypi.py

```diff
@@ -298,7 +300,9 @@ def get_simplelinks_perstage(self, project):
         """
         project = normalize_name(project)
         is_fresh, links, cache_serial = self._load_cache_links(project)
-        if is_fresh:
+        if is_fresh or self.offline:
```

Comment:
I do not understand test.offline being tested twice.
This might ne to be refactored to better express intent.

Reply:
This is changed in the 404 handling commit later on